pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -82,147 +85,87 @@ func watchChaosResources(handlerForOwner handler.EnqueueRequestForOwner, c contr
 func (r *ReconcileChaosEngine) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling ChaosEngine")
-
 	// Fetch the ChaosEngine instance
-	instance := &litmuschaosv1alpha1.ChaosEngine{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
-	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			return reconcile.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
+	err := r.getChaosEngineInstance(request)
+	if err.Error() != "No error" {
 		return reconcile.Result{}, err
 	}
-	var engine engineInfo
-	engine.instance = instance
 	// Fetch the app details from ChaosEngine instance. Check if app is present
 	// Also check, if the app is annotated for chaos & that the labels are unique
 
 	// TODO: Get app kind from chaosengine spec as well. Using "deploy" for now
 	// TODO: Freeze label format in chaosengine( "=" as a const)
-	appInfo := &applicationInfo{}
-	appInfo, err = appInfo.initializeApplicationInfo(instance)
+	err = getApplicationDetail()
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	engine.appInfo = appInfo
-
-	var appExperiments []string
-	for _, exp := range appInfo.experimentList {
-		appExperiments = append(appExperiments, exp.Name)
-	}
-	engine.appExperiments = appExperiments
-
-	log.Info("App key derived from chaosengine is ", "appLabelKey", appLabelKey)
-	log.Info("App Label derived from Chaosengine is ", "appLabelValue", appLabelValue)
-	log.Info("App NS derived from Chaosengine is ", "appNamespace", appInfo.namespace)
-	log.Info("Exp list derived from chaosengine is ", "appExpirements", appExperiments)
-	log.Info("Monitoring Status derived from chaosengine is", "monitoringstatus", engine.instance.Spec.Monitoring)
-
 	// Use client-Go to obtain a list of apps w/ specified labels
-	restConfig, err := config.GetConfig()
+	clientSet, err := createClientSet()
 	if err != nil {
-		log.Error(err, "unable to get rest kube config")
```

Comment:
Change this line to `log.Error("Clientset generation failed with error:  ", err)`